### ISSUE TITLE
no-redeclare does not work with renaming in destructures inside for-loop

### ISSUE THREAD

**Issue by zachdini** [eslint/eslint#7917]
Eslint appears to not catch no-redeclare if I rename a variable when destructuring within a for-loop. The code below does not throw an error:

```js
function foo(params) {
    for (let i = 0; i < params.length; ++i) {
        const { value: paramValue } = params[i];
        const paramValue = params[i].value;
    }
}
```

I'm running:
eslint: v3.13.1
node: v6.9.1
npm: 3.10.8

**Comment by eslintbot** [comment]
Hi @zachdini, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by nzakas** [comment]
Not enough information to evaluate, so closing.
